include/seqan3/alignment/matrix/detail/score_matrix_single_column.hpp

```diff
@@ -0,0 +1,254 @@
+     * the number of rows is set to the size of sequence2 plus 1 for the initialisation as well.
+     * Reallocation happens only if the new column size exceeds the current capacity of the optimal and horizontal
+     * score column. The underlying vectors will not be cleared during the reset.
+     */
+    template <std::ranges::forward_range sequence1_t, std::ranges::forward_range sequence2_t>
+    void reset_matrix(sequence1_t && sequence1, sequence2_t && sequence2)
+    {
+        m_number_of_columns = std::ranges::distance(sequence1) + 1;
+        size_t number_of_rows = std::ranges::distance(sequence2) + 1;
+        m_optimal_column.resize(number_of_rows);
+        m_horizontal_column.resize(number_of_rows);
+        m_vertical_column = views::repeat_n(score_t{}, number_of_rows);
+    }
+
+    /*!\name Iterators
+     * \{
+     */
+    //!\brief Returns the iterator pointing to the first column.
+    matrix_iterator begin()
+    {
+        return matrix_iterator{*this};
+    }
+    //!\brief This score matrix is not const-iterable.
+    matrix_iterator begin() const = delete;
+
+    //!\brief Returns a default sentinel indicating the end of the matrix.
+    std::ranges::default_sentinel_t end()
+    {
+        return std::ranges::default_sentinel;
+    }
+
+    //!\brief This score matrix is not const-iterable.
+    std::ranges::default_sentinel_t end() const = delete;
+    //!\}
+};
+
+/*!\brief Score matrix for the pairwise alignment using only a single column.
+ * \implements std::input_iterator
+ *
+ * \details
+ *
+ * Implements a counted iterator to simulate the iteration over the actual matrix. When dereferenced, the
+ * iterator returns a view over the allocated memory of the respective columns. The returned view zips
+ * the three columns into a single range and transforms the returned tuple to a
+ * seqan3::detail::affine_cell_proxy to simplify the access to the correct values without knowing the internal
+ * tuple layout returned by the seqan3::views::zip view.
+ */
+template <typename score_t>
+class score_matrix_single_column<score_t>::matrix_iterator
+{
+private:
+
+    //!\brief The type of the zipped score column.
+    using matrix_column_t = decltype(views::zip(std::declval<physical_column_t &>(),
+                                                std::declval<physical_column_t &>(),
+                                                std::declval<virtual_column_t &>()));
+
+    //!\brief The transform adaptor to convert the tuple from the zip view into a seqan3::detail::affine_cell_type.
+    static constexpr auto transform_to_affine_cell = std::views::transform([] (auto && tpl)
+    {
+        using fwd_tuple_t = decltype(tpl);
+        return affine_cell_proxy<remove_cvref_t<fwd_tuple_t>>{std::forward<fwd_tuple_t>(tpl)};
+    });
+
+    //!\brief The pointer to the underlying matrix.
+    score_matrix_single_column * m_host_ptr{nullptr};
+    //!\brief The current column index.
+    size_t m_current_column{};
```

Comment:
```suggestion
    size_t current_column{};
```
Or will the naming change somwhat later?
(same for the other members, for me the `m_` prefix everywhere does not help much)

Reply:
Yes, it helps a lot in many cases. And there is no rule about it. So we need a rule. That's my proposal. :)